Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -414,7 +414,192 @@ public void ForceRefresh()
+            {   // Nothing to do here.
+                return;
+            }
+
+            SerializedObject configObject = new SerializedObject(this);
+            SerializedProperty iconStyleProp = configObject.FindProperty("iconStyle");
+            SerializedProperty iconSetProp = configObject.FindProperty("iconSet");
+            SerializedProperty iconQuadTextureProp = configObject.FindProperty("iconQuadTexture");
+
+            if (iconQuadRenderer.gameObject.activeSelf && !iconQuadRenderer.enabled)
+            {   // If the quad renderer is disabled, enable it and disable the quad renderer game object instead.
+                // Disabling the quad renderer used to be the preferred way to disable icons but it's no longer consistent with our icon style.
+                iconQuadRenderer.enabled = true;
+                iconQuadRenderer.gameObject.SetActive(false);
+                EditorUtility.SetDirty(gameObject);
+            }
+
+            if (!iconQuadRenderer.gameObject.activeSelf && !iconSpriteRenderer.gameObject.activeSelf && !iconCharLabel.gameObject.activeSelf)
+            {   // If all the icon objects are disabled, set the icon style to none and do nothing else.
+                iconStyleProp.enumValueIndex = (int)ButtonIconStyle.None;
+                configObject.ApplyModifiedProperties();
+                EditorUtility.SetDirty(gameObject);
+                return;
+            }
+
+            if (iconSet != defaultIconSet)
+            {  // This button is using a custom icon set, so we can't assume material differences mean it needs an upgrade.
+                return;
+            }
+
+            if (iconQuadRenderer.sharedMaterial == defaultButtonQuadMaterial) 
+            {   // This button is using the default material, so it's not a customized button.
+                return;
+            }
+
+            string assetPath = AssetDatabase.GetAssetPath(iconQuadRenderer.sharedMaterial);
+            if (string.IsNullOrEmpty(assetPath))
+            {   // If the asset path is null, this material instance exists only in memory.
+                return;
+            }
+
+            Material targetQuadMaterial = iconQuadRenderer.sharedMaterial;
+            Texture targetQuadIcon = targetQuadMaterial.mainTexture;
+
+            if (targetQuadIcon == null)
+            {   // There is no icon to copy.
+                return;
+            }
+
+            ButtonIconSet targetIconSet = iconSet;
+            bool createdIconSet = false;
+            string generatedIconSetFolder = System.IO.Path.Combine(MixedRealityToolkitFiles.GetGeneratedFolder, customIconSetsFolderName);
+
+            // If this icon set doesn't have our icon in it, we need to either add it or create a new icon set
+            if (!iconSet.TryGetQuadIcon(targetQuadIcon.name, out Texture2D quadIcon))
+            {
+                if (iconSet == defaultIconSet)
```

Comment:
Trying to write unit tests, and this check seems unnecessary because of this line 447

`if (iconSet != defaultIconSet)`

It's a bit hard to test this case since this tool ends up creating or modifying the custom icon set if its successful. Thoughts?